cpp/oneapi/dal/backend/primitives/ndarray.hpp

```diff
@@ -265,10 +278,30 @@ class ndview : public ndarray_base<axis_count, order> {
             auto* const row = get_mutable_data() + id0 * this->get_stride(0);
             return *(row + id1);
         }
-        auto* const col = get_mutable_data() + id1 * this->get_stride(1);
-        return *(col + id0);
+        else {
+            auto* const col = get_mutable_data() + id1 * this->get_stride(1);
+            return *(col + id0);
+        }
     }
 
+#ifdef ONEDAL_DATA_PARALLEL
+    template <std::int64_t n = axis_count, typename = std::enable_if_t<n == 1>>
+    T at_device(sycl::queue& queue, std::int64_t id, const event_vector& deps = {}) const {
+        return this->get_slice(id, id + 1).to_host(queue, deps).at(0);
+    }
+
+    template <std::int64_t n = axis_count, typename = std::enable_if_t<n == 2>>
+    T at_device(sycl::queue& queue,
```

Comment:
Those look like not the performance-oriented APIs. Do we really need those?
My concern is that we can get sub-optimal performance due to improper use of those APIs.